state-chain/runtime/src/event_encoder.rs

```diff
@@ -0,0 +1,111 @@
+use codec::{Decode, Encode};
+use scale_info::{prelude::string::String, PortableRegistry};
+use serde::{Deserialize, Serialize};
+use sp_std::vec::Vec;
+
+use crate::RuntimeEvent;
+
+#[derive(Encode, Decode, Clone, Copy, Eq, PartialEq, Serialize, Deserialize, Debug)]
+pub enum EventDecoderError {
+	FailedToDecodeFromScaleBytes,
+	FailedToDecodeFromString,
+	FailedToConvertIntoJson,
+}
+
+/// A struct that provides interface for decoding RuntimeEvents.
+pub struct RuntimeEventDecoder {
+	registry: PortableRegistry,
+	type_id: u32,
+}
+
+impl RuntimeEventDecoder {
```

Comment:
This type can be moved out of the runtime - can be in the custom rpc crate I guess. 

I would also suggest making this a `RuntimeTypeDecoder<T: TypeInfo>`. This should make it easier to test with arbitrary values.

Reply:
Moved this to a mod under the custom-rpc module.